### PR TITLE
account for memory allocated during text analysis

### PR DESCRIPTION
**Pull request by mschoch**

### CODE REVIEW

File: analysis/freq.go [hunk 1]
```diff
@@ -34,6 +59,14 @@ type TokenFreq struct {
 	frequency int
 }
 
+func (tf *TokenFreq) Size() int {
+	rv := reflectStaticSizeTokenFreq
```
Comment (abhinavdangeti) [inline analysis/freq.go]: Shouldn't we also include len(tf.Term) here?

Comment (mschoch) [reply analysis/freq.go]: ya good catch

File: index/scorch/scorch.go [hunk 2]
```diff
@@ -295,12 +297,17 @@ func (s *Scorch) Batch(batch *index.Batch) (err error) {
 	// wait for analysis result
 	analysisResults := make([]*index.AnalysisResult, int(numUpdates))
 	var itemsDeQueued uint64
+	var totalAnalysisSize int
 	for itemsDeQueued < numUpdates {
 		result := <-resultChan
+		resultSize := result.Size()
+		atomic.AddUint64(&s.iStats.newSegBufBytesAdded, uint64(resultSize))
```
Comment (abhinavdangeti) [inline index/scorch/scorch.go]: Sorry for missing this before, but shouldn't this have been analysisBytesAdded, and below analysisBytesRemoved?

**Comment by mschoch** [comment]
Did an index build with these changes, seemed better, but not still not great.

**Pull request merged**